src/drivers/amd/renoir/ipc.c

```diff
@@ -174,9 +205,10 @@ int ipc_platform_send_msg(struct ipc_msg *msg)
 	acp_dsp_to_host_Intr_trig();
 	/* Disable the trigger bit in ACP_DSP_SW_INTR_TRIG register */
 	sw_intr_trig = (acp_sw_intr_trig_t)io_reg_read(PU_REGISTER_BASE + ACP_SW_INTR_TRIG);
-	sw_intr_trig.bits.trig_host_to_dsp0_intr1 = INTERRUPT_DISABLE;
 	sw_intr_trig.bits.trig_dsp0_to_host_intr = INTERRUPT_DISABLE;
 	io_reg_write((PU_REGISTER_BASE + ACP_SW_INTR_TRIG), sw_intr_trig.u32all);
+	io_reg_write((PU_REGISTER_BASE + ACP_AXI2DAGB_SEM_0), lock);
```

Comment:
you have a loop on read but not on write, is this intentional?

Reply:
yes. The read requires loop based on hw behavior.